utils/merge/mergeSchemas.js

```diff
@@ -34,12 +34,31 @@ module.exports = function mergeSchemas(schema1, schema2) {
       },
     },
   ];
+  const propsToOverride = [
+    {
+      name: 'required',
+      action(first, second) {
+        if (!schema1.properties || !schema2.properties) return first;
+
+        const firstProperties = Object.keys(schema1.properties);
```

Comment:
To get values of `required` prop we use `result` (resulting schema after merge) and schema 2 (`result[key], schema2[key]`). Here we use original schema 1 and schema 2.

Is this intended?

And, btw, maybe it will be better to pass schema1 and schema 2 as `action` func args and not to rely on closure? Then we will be able to move `propsToMerge` and `propsToOverride` out of function and not to generate new consts on every function call. Up to you, as one says, but worth considering.